src/main/python/systemds/script_building/dag.py

```diff
@@ -74,12 +74,13 @@ class DAGNode(ABC):
     sds_context: 'SystemDSContext'
     _unnamed_input_nodes: Sequence[Union['DAGNode', str, int, float, bool]]
     _named_input_nodes: Dict[str, Union['DAGNode', str, int, float, bool]]
+    _named_output_nodes: Dict[str, Union['DAGNode', str, int, float, bool]]
```

Comment:
Without this we won't be able to do something like this: `node2 = node1["res"].abs()` or this `node3 = node2["X_train"] + node2["Y_train"]`? The only possibility would be to get the output node by the index. (While at the moment we can do both, by index (not included in this PR, but already implemented) or by name). Or am I missing something?

Reply:
Maybe my comment was not clear.

The issue is that the list can only be name or number based output nodes, not both. Therefore making two constructors enforce this design, one that specify a output list length and one with a list of names. That is why my first comment says hide it away, because with the previous design, one could access both arguments, and the implementation therefore have to throw an error if both arguments are given.
If you make two constructors then no exception handling is needed because wrong arguments can not happen.

I will change the API appropriately while merging, so no need to change anything.